### PR TITLE
fix fp8: exclude MoE router linears from torchao fp8 conversion

### PR DESCRIPTION
**Pull request by ved1beta**
# Description
fp8 training converted every nn.Linear except first/last to Float8Linear, including the MoE router gate tensorwise-fp8 router logits destabilize softmax/topk routing and diverge to NaN. This patches accelerate's fp8 conversion to skip router/gate linears while keeping experts and attention in fp8.
## Motivation and Context
#3077

## How has this been tested?
runs fine without nan 

## AI Usage Disclaimer
fable 5 helped T_T" 

## Summary by CodeRabbit

* **New Features**
  * FP8 training now better supports Mixture-of-Experts models by automatically excluding router/gate layers from FP8 conversion.

* **Bug Fixes**
  * Improved FP8 handling to reduce the risk of numerical issues or unexpected behavior when using expert-routing architectures.
  * FP8 setup now applies the router exclusion consistently alongside existing FP8 configuration.

### CODE REVIEW

File: src/axolotl/loaders/patch_manager.py [hunk 1]
```diff
@@ -632,13 +632,17 @@ def _fix_nemotron_h_conversion_mapping():
     def _apply_fp8_patches(self):
         """Apply patches for FP8 support."""
         if self.cfg.fp8:
+            from axolotl.monkeypatch.accelerate.float8_moe_filter import (
```
Comment (winglian) [inline src/axolotl/loaders/patch_manager.py:630]: was this tested? this import is incorrect.

Comment (ved1beta) [reply src/axolotl/loaders/patch_manager.py:630]: renamed/ moved the file after changes , that caused mismatch  (100% tested 🫡 )
